PennMobile/Dining/SwiftUI/DiningAnalyticsViewModel.swift

```diff
@@ -67,36 +68,46 @@ class DiningAnalyticsViewModel: ObservableObject {
                 return
             }
-            self.swipesPredictedZeroDate = self.predictZeroDate(firstBalance: firstSwipeBalance, lastBalance: lastSwipeBalance)
-            self.predictedSwipesSemesterEndBalance = self.predictSemesterEndBalance(firstBalance: firstSwipeBalance, lastBalance: lastSwipeBalance)
+            let dollarPredictions = self.getPredictions(firstBalance: maxDollarBalance, lastBalance: lastDollarBalance)
+            self.dollarSlope = dollarPredictions.slope
+            self.dollarPredictedZeroDate = dollarPredictions.predictedZeroDate
+            self.predictedDollarSemesterEndBalance = dollarPredictions.predictedEndBalance
+            self.dollarAxisLabel = self.getAxisLabelsYX(from: self.dollarHistory)
+            let swipePredictions = self.getPredictions(firstBalance: maxSwipeBalance, lastBalance: lastSwipeBalance)
+            self.swipeSlope = swipePredictions.slope
+            self.swipesPredictedZeroDate = swipePredictions.predictedZeroDate
+            self.predictedSwipesSemesterEndBalance = swipePredictions.predictedEndBalance
             self.swipeAxisLabel = self.getAxisLabelsYX(from: self.swipeHistory)
-            self.predictedSwipesZeroPoint = PredictionsGraphView.getPredictionZeroPoint(startOfSemester: Date.startOfSemester, endOfSemester: Date.endOfSemester, predictedZeroDate: self.swipesPredictedZeroDate)
         }
     }
-    func predictZeroDate(firstBalance: DiningAnalyticsBalance, lastBalance: DiningAnalyticsBalance) -> Date {
-        let yIntercept = firstBalance.balance
-        let slope = (lastBalance.balance - firstBalance.balance) / Double(Calendar.current.dateComponents([.day], from: firstBalance.date, to: lastBalance.date).day!)
-        let offset = -yIntercept / slope
-        var dateComponent = DateComponents()
-        dateComponent.day = Int(offset)
-        return Calendar.current.date(byAdding: dateComponent, to: firstBalance.date)!
-    }
-    func predictSemesterEndBalance(firstBalance: DiningAnalyticsBalance, lastBalance: DiningAnalyticsBalance) -> Double {
-        // y = slope * x + yIntercept
-        let yIntercept = firstBalance.balance
-        let slope = (lastBalance.balance - firstBalance.balance) / Double(Calendar.current.dateComponents([.day], from: firstBalance.date, to: lastBalance.date).day!)
-        let diffInDays = Calendar.current.dateComponents([.day], from: firstBalance.date, to: Date.endOfSemester).day!
-        return ((slope * Double(diffInDays)) + yIntercept)
+    func getPredictions(firstBalance: DiningAnalyticsBalance, lastBalance: DiningAnalyticsBalance) -> (slope: Double, predictedZeroDate: Date, predictedEndBalance: Double) {
+        if firstBalance.date == lastBalance.date {
+            let zeroDate = Calendar.current.date(byAdding: .day, value: 1, to: Date.endOfSemester)!
+            return (Double.infinity, zeroDate, lastBalance.balance)
```

Comment:
Curious why we're returning slope as infinity in this case. Maybe mathematically it is more accurate, but it seems to be treated the same as 0 in PredictionSlopePath, and I think in the context of predictions it makes sense for 0 to be the max slope. For that reason, I think you can get rid of this `if` statement altogether, as it should get caught in the `firstBalance.balance == lastBalance.balance` case. Please let me know if you agree!

Reply:
Yea I think that makes sense. I'll combine these two cases together and just return 0.0.